### PR TITLE
Re-enable MPI + setup buildkite

### LINKED ISSUE #127: extend `write_exodus` to deal with distributed Topology
We need to extend the ClimaCoreTempestRemap function:

```
write_exodus(::Any, !Matched::ClimaCore.Topologies.Topology2D; normalize_coordinates) 
```
to

```
write_exodus(::String, ::ClimaCore.Topologies.DistributedTopology2D{ClimaCommsMPI.MPICommsContext, ClimaCore.Meshes.EquiangularCubedSphere{ClimaCore.Domains.SphereDomain{Float32}, ClimaCore.Meshes.NormalizedBilinearMap}, Vector{CartesianIndex{3}}, Array{Int64, 3}, NamedTuple{(), Tuple{}}})
```
so that we can set boundary conditions from files onto the distributed topology. The easiest solution is to create a surrogate topology which is not distributed, collect data and distribute onto the model (distributed) topology.

### PR DESCRIPTION
**Pull request by LenkaNovak**
## Purpose 

#74 (addresses remaining high-priority bullet point)

## To-do + Content

- [x] fix Atmos MPI race condition
- [x] revamp bcfile_reader to handle MPI (leveraging parallel HDF5) and closes #127 
- [x] bring tests up to date
- ~~[ ] add MPI buildkite at high resolution~~ separate PR after ClimaTimeSteppers integrated

## Review checklist

- [x] I have:
  - followed the codebase contribution guide: https://clima.github.io/ClimateMachine.jl/latest/Contributing/
  - followed the style guide: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/
  - followed the documentation policy: https://github.com/CliMA/policies/wiki/Documentation-Policy
  - checked that this PR does not duplicate an open PR.

- [x] In the Content, I have included 
  - relevant unit tests, and integration tests, 
  - appropriate docstrings on all functions, structs, and modules, and included relevent documentation.

### CODE REVIEW

File: experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl [hunk 1]
```diff
@@ -53,41 +56,35 @@ function bcfile_info_init(
...
-        datafile_rll,
-        varname,
-        boundary_space,
-        outfile = varname * "_cgll.g",
-        mono = mono,
-    )
-    ds = Dataset(datafile_cgll, "r")
+    # regrid all times and save to hdf5 files
+    hd_outfile_root = varname * "_cgll"
+    if !is_distributed || ClimaComms.iamroot(comms_ctx)
```
Comment (simonbyrne) [inline experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl:61]: This should work generally:
```suggestion
    if ClimaComms.iamroot(comms_ctx)
```

File: experiments/AMIP/moist_mpi_earth/cli_options.jl [hunk 2]
```diff
@@ -370,3 +374,6 @@ function time_to_seconds(s::String)
     end
     error("Uncaught case in computing time from given string.")
 end
+
+import Base.joinpath
+Base.joinpath(s::String, ::Nothing) = s
```
Comment (simonbyrne) [inline experiments/AMIP/moist_mpi_earth/cli_options.jl:379]: why is this needed? generally not a good idea to type-pirate Base functions.

Comment (LenkaNovak) [reply experiments/AMIP/moist_mpi_earth/cli_options.jl:379]: Hehehe! 🏴‍☠️ This was to enable the run name cli arg to be `nothing` by default, but if this is not the best practice, I can make the default a string.

Comment (simonbyrne) [reply experiments/AMIP/moist_mpi_earth/cli_options.jl:379]: yes, that would be a better idea.

**Comment by LenkaNovak** [comment]
(approved on Slack)
